# Student Performance Data Analysis


In [1]:
import numpy as np
import pandas as pd


In [5]:
# loading the data set we are working on 

df = pd.read_csv("../data/student_prerformance.csv")
df

student_id  age gender academic_level  study_hours  self_study_hours  \
0              1   18  Other    High School         7.64              1.56   
1              2   18  Other    High School         2.21              2.22   
2              3   22   Male    High School         3.45              0.00   
3              4   17  Other    High School         5.75              2.08   
4              5   19  Other    High School         6.83              1.72   
...          ...  ...    ...            ...          ...               ...   
4995        4996   20  Other  Undergraduate         3.93              1.42   
4996        4997   24   Male    High School         4.00              0.00   
4997        4998   22   Male  Undergraduate         5.63              2.52   
4998        4999   17   Male  Undergraduate         3.58              2.83   
4999        5000   22  Other  Undergraduate         5.60              4.14   

      online_classes_hours  social_media_hours  gaming_hours  sleep_hours  \
0                     2.20                3.05          2.19         6.52   
1                     2.10                1.65          2.55         5.97   
2                     0.29                1.34          2.08         8.39   
3                     3.01                2.27          2.20         6.31   
4                     3.33                2.65          0.70         8.01   
...                    ...                 ...           ...          ...   
4995                  0.54                2.58          2.62         9.30   
4996                  3.00                4.45          2.47         6.42   
4997                  2.89                3.58          1.07         6.65   
4998                  1.53                5.88          1.28         7.36   
4999                  2.17                3.05          2.74         6.38   

      ...  exercise_minutes  caffeine_intake_mg  part_time_job  \
0     ...                81                  38              1   
1     ...               111                 339              0   
2     ...                68                 266              0   
3     ...               113                 480              1   
4     ...               121                  24              1   
...   ...               ...                 ...            ...   
4995  ...               113                 496              0   
4996  ...               146                 411              0   
4997  ...               123                 306              0   
4998  ...                54                 304              1   
4999  ...                24                 219              0   

      upcoming_deadline  internet_quality mental_health_score  focus_index  \
0                     0              Good                  10        43.05   
1                     0              Good                   3        15.92   
2                     1              Good                   8        27.39   
3                     1              Poor                   3        22.31   
4                     0              Good                   8        38.11   
...                 ...               ...                 ...          ...   
4995                  0              Good                   9        35.55   
4996                  1              Poor                   5        16.87   
4997                  1              Good                   7        23.85   
4998                  1              Good                   4        13.04   
4999                  1              Good                   3        29.98   

      burnout_level  productivity_score  exam_score  
0             31.77               73.65       50.16  
1             37.00               13.70        1.00  
2             34.37               45.15       18.30  
3             77.31               20.92        9.37  
4             39.53               59.23       27.81  
...             ...                 ...         ...  
4995          33.24               48

In [6]:
df.shape

(5000, 21)

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   student_id            5000 non-null   int64  
 1   age                   5000 non-null   int64  
 2   gender                5000 non-null   object 
 3   academic_level        5000 non-null   object 
 4   study_hours           5000 non-null   float64
 5   self_study_hours      5000 non-null   float64
 6   online_classes_hours  5000 non-null   float64
 7   social_media_hours    5000 non-null   float64
 8   gaming_hours          5000 non-null   float64
 9   sleep_hours           5000 non-null   float64
 10  screen_time_hours     5000 non-null   float64
 11  exercise_minutes      5000 non-null   int64  
 12  caffeine_intake_mg    5000 non-null   int64  
 13  part_time_job         5000 non-null   int64  
 14  upcoming_deadline     5000 non-null   int64  
 15  internet_quality     

In [8]:
df.describe().T

count         mean          std   min        25%  \
student_id            5000.0  2500.500000  1443.520003   1.0  1250.7500   
age                   5000.0    20.520400     2.870406  16.0    18.0000   
study_hours           5000.0     4.539594     1.821665   0.0     3.2500   
self_study_hours      5000.0     2.478734     1.177990   0.0     1.6600   
online_classes_hours  5000.0     2.011984     0.983906   0.0     1.3200   
social_media_hours    5000.0     2.998086     1.467949   0.0     1.9900   
gaming_hours          5000.0     1.564514     1.110807   0.0     0.6700   
sleep_hours           5000.0     7.016492     1.163692   4.0     6.2375   
screen_time_hours     5000.0     6.979588     2.486214   1.0     5.2800   
exercise_minutes      5000.0    74.535600    42.932293   0.0    37.0000   
caffeine_intake_mg    5000.0   251.450400   143.842712   0.0   129.0000   
part_time_job         5000.0     0.498200     0.500047   0.0     0.0000   
upcoming_deadline     5000.0     0.501400     0.500048   0.0     0.0000   
mental_health_score   5000.0     5.507400     2.869145   1.0     3.0000   
focus_index           5000.0    29.431616     9.962902   1.0    22.5675   
burnout_level         5000.0    45.615324    14.246591   1.0    35.7275   
productivity_score    5000.0    37.267716    16.849397   1.0    25.2900   
exam_score            5000.0    18.803752    12.130840   1.0     9.3375   

                          50%        75%      max  
student_id            2500.50  3750.2500  5000.00  
age                     20.00    23.0000    25.00  
study_hours              4.53     5.7600    11.84  
self_study_hours         2.48     3.2900     7.41  
online_classes_hours     2.01     2.6900     6.00  
social_media_hours       2.98     4.0300     8.28  
gaming_hours             1.49     2.3400     5.64  
sleep_hours              7.01     7.8100    10.00  
screen_time_hours        6.95     8.7100    15.30  
exercise_minutes        75.00   112.0000   149.00  
caffeine_intake_mg     252.00   376.0000   499.00  
part_time_job            0.00     1.0000     1.00  
upcoming_deadline        1.00     1.0000     1.00  
mental_health_score      5.00     8.0000    10.00  
focus_index             29.43    36.2425    63.48  
burnout_level           45.69    55.3525    97.58  
productivity_score      36.86    49.1425    98.02  
exam_score              18.01    27.4000    64.09

In [10]:
df.columns

Index(['student_id', 'age', 'gender', 'academic_level', 'study_hours',
       'self_study_hours', 'online_classes_hours', 'social_media_hours',
       'gaming_hours', 'sleep_hours', 'screen_time_hours', 'exercise_minutes',
       'caffeine_intake_mg', 'part_time_job', 'upcoming_deadline',
       'internet_quality', 'mental_health_score', 'focus_index',
       'burnout_level', 'productivity_score', 'exam_score'],
      dtype='object')

In [11]:
df.head()

student_id  age gender academic_level  study_hours  self_study_hours  \
0           1   18  Other    High School         7.64              1.56   
1           2   18  Other    High School         2.21              2.22   
2           3   22   Male    High School         3.45              0.00   
3           4   17  Other    High School         5.75              2.08   
4           5   19  Other    High School         6.83              1.72   

   online_classes_hours  social_media_hours  gaming_hours  sleep_hours  ...  \
0                  2.20                3.05          2.19         6.52  ...   
1                  2.10                1.65          2.55         5.97  ...   
2                  0.29                1.34          2.08         8.39  ...   
3                  3.01                2.27          2.20         6.31  ...   
4                  3.33                2.65          0.70         8.01  ...   

   exercise_minutes  caffeine_intake_mg  part_time_job  upcoming_deadline  \
0                81                  38              1                  0   
1               111                 339              0                  0   
2                68                 266              0                  1   
3               113                 480              1                  1   
4               121                  24              1                  0   

   internet_quality mental_health_score  focus_index  burnout_level  \
0              Good                  10        43.05          31.77   
1              Good                   3        15.92          37.00   
2              Good                   8        27.39          34.37   
3              Poor                   3        22.31          77.31   
4              Good                   8        38.11          39.53   

   productivity_score  exam_score  
0               73.65       50.16  
1               13.70        1.00  
2               45.15       18.30  
3               20.92        9.37  
4               59.23       27.81  

[5 rows x 21 columns]

In [ ]:
df.describe(include="object").T #df.describe()T help to distribute the values using include method 

count unique           top  freq
gender            5000      3          Male  1719
academic_level    5000      3  Postgraduate  1687
internet_quality  5000      3          Good  1722

Here, the gender count is = 5000 with 3 unique value with Topo male gender with frequence 1719.

academic_level -> top Postgraduate with freq 1687.

internet_quality -> good with feq 1722.

In [14]:
df.describe(include="float").T

count       mean        std  min      25%    50%  \
study_hours           5000.0   4.539594   1.821665  0.0   3.2500   4.53   
self_study_hours      5000.0   2.478734   1.177990  0.0   1.6600   2.48   
online_classes_hours  5000.0   2.011984   0.983906  0.0   1.3200   2.01   
social_media_hours    5000.0   2.998086   1.467949  0.0   1.9900   2.98   
gaming_hours          5000.0   1.564514   1.110807  0.0   0.6700   1.49   
sleep_hours           5000.0   7.016492   1.163692  4.0   6.2375   7.01   
screen_time_hours     5000.0   6.979588   2.486214  1.0   5.2800   6.95   
focus_index           5000.0  29.431616   9.962902  1.0  22.5675  29.43   
burnout_level         5000.0  45.615324  14.246591  1.0  35.7275  45.69   
productivity_score    5000.0  37.267716  16.849397  1.0  25.2900  36.86   
exam_score            5000.0  18.803752  12.130840  1.0   9.3375  18.01   

                          75%    max  
study_hours            5.7600  11.84  
self_study_hours       3.2900   7.41  
online_classes_hours   2.6900   6.00  
social_media_hours     4.0300   8.28  
gaming_hours           2.3400   5.64  
sleep_hours            7.8100  10.00  
screen_time_hours      8.7100  15.30  
focus_index           36.2425  63.48  
burnout_level         55.3525  97.58  
productivity_score    49.1425  98.02  
exam_score            27.4000  64.09

The max Study_hours of a student is 11.84 % 

student have highest Producivity_score above 95% 

In [18]:
df.describe(include="int").T

count       mean          std   min      25%     50%  \
student_id           5000.0  2500.5000  1443.520003   1.0  1250.75  2500.5   
age                  5000.0    20.5204     2.870406  16.0    18.00    20.0   
exercise_minutes     5000.0    74.5356    42.932293   0.0    37.00    75.0   
caffeine_intake_mg   5000.0   251.4504   143.842712   0.0   129.00   252.0   
part_time_job        5000.0     0.4982     0.500047   0.0     0.00     0.0   
upcoming_deadline    5000.0     0.5014     0.500048   0.0     0.00     1.0   
mental_health_score  5000.0     5.5074     2.869145   1.0     3.00     5.0   

                         75%     max  
student_id           3750.25  5000.0  
age                    23.00    25.0  
exercise_minutes      112.00   149.0  
caffeine_intake_mg    376.00   499.0  
part_time_job           1.00     1.0  
upcoming_deadline       1.00     1.0  
mental_health_score     8.00    10.0

In [19]:
df.isnull().sum()

student_id              0
age                     0
gender                  0
academic_level          0
study_hours             0
self_study_hours        0
online_classes_hours    0
social_media_hours      0
gaming_hours            0
sleep_hours             0
screen_time_hours       0
exercise_minutes        0
caffeine_intake_mg      0
part_time_job           0
upcoming_deadline       0
internet_quality        0
mental_health_score     0
focus_index             0
burnout_level           0
productivity_score      0
exam_score              0
dtype: int64

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   student_id            5000 non-null   int64  
 1   age                   5000 non-null   int64  
 2   gender                5000 non-null   object 
 3   academic_level        5000 non-null   object 
 4   study_hours           5000 non-null   float64
 5   self_study_hours      5000 non-null   float64
 6   online_classes_hours  5000 non-null   float64
 7   social_media_hours    5000 non-null   float64
 8   gaming_hours          5000 non-null   float64
 9   sleep_hours           5000 non-null   float64
 10  screen_time_hours     5000 non-null   float64
 11  exercise_minutes      5000 non-null   int64  
 12  caffeine_intake_mg    5000 non-null   int64  
 13  part_time_job         5000 non-null   int64  
 14  upcoming_deadline     5000 non-null   int64  
 15  internet_quality     

In [ ]:

df.describe()

student_id          age  study_hours  self_study_hours  \
count  5000.000000  5000.000000  5000.000000       5000.000000   
mean   2500.500000    20.520400     4.539594          2.478734   
std    1443.520003     2.870406     1.821665          1.177990   
min       1.000000    16.000000     0.000000          0.000000   
25%    1250.750000    18.000000     3.250000          1.660000   
50%    2500.500000    20.000000     4.530000          2.480000   
75%    3750.250000    23.000000     5.760000          3.290000   
max    5000.000000    25.000000    11.840000          7.410000   

       online_classes_hours  social_media_hours  gaming_hours  sleep_hours  \
count           5000.000000         5000.000000   5000.000000  5000.000000   
mean               2.011984            2.998086      1.564514     7.016492   
std                0.983906            1.467949      1.110807     1.163692   
min                0.000000            0.000000      0.000000     4.000000   
25%                1.320000            1.990000      0.670000     6.237500   
50%                2.010000            2.980000      1.490000     7.010000   
75%                2.690000            4.030000      2.340000     7.810000   
max                6.000000            8.280000      5.640000    10.000000   

       screen_time_hours  exercise_minutes  caffeine_intake_mg  part_time_job  \
count        5000.000000       5000.000000         5000.000000    5000.000000   
mean            6.979588         74.535600          251.450400       0.498200   
std             2.486214         42.932293          143.842712       0.500047   
min             1.000000          0.000000            0.000000       0.000000   
25%             5.280000         37.000000          129.000000       0.000000   
50%             6.950000         75.000000          252.000000       0.000000   
75%             8.710000        112.000000          376.000000       1.000000   
max            15.300000        149.000000          499.000000       1.000000   

       upcoming_deadline  mental_health_score  focus_index  burnout_level  \
count        5000.000000          5000.000000  5000.000000    5000.000000   
mean            0.501400             5.507400    29.431616      45.615324   
std             0.500048             2.869145     9.962902      14.246591   
min             0.000000             1.000000     1.000000       1.000000   
25%             0.000000             3.000000    22.567500      35.727500   
50%             1.000000             5.000000    29.430000      45.690000   
75%             1.000000             8.000000    36.242500      55.352500   
max             1.000000            10.000000    63.480000      97.580000   

       productivity_score   exam_score  
count         5000.000000  5000.000000  
mean            37.267716    18.803752  
std             16.849397    12.130840  
min              1.000000     1.000000  
25%             25.290000     9.337500  
50%             36.860000    18.010000  
75%             49.142500    27.400000  
max             98.020000    64.090000